src/api/publicationSpecsExt.ts

```diff
@@ -27,3 +27,16 @@ export const getPublicationHistoryByCatalogName = (catalogName: string) => {
             ascending: false,
         });
 };
+
+export const getLiveSpecIdByPublication = (
+    pubId: string | null, // Do not actually pass null... just making typing easiser
+    entityType: Entity
+) => {
+    return supabaseClient
+        .from<PublicationSpecsExt_PublicationHistory>(
+            TABLES.PUBLICATION_SPECS_EXT
+        )
+        .select(`live_spec_id`)
+        .eq('pub_id', pubId)
+        .eq('spec_type', entityType);
```

Comment:
The combination of `pub_id` and `spec_type` isn't enough to uniquely identify a publication spec. Consider the `pub_id` similar to a `draft_id`, in that it can identify any number or combination of specs. So you could have multiple captures in a publication, for example. This seems like an issue because the callsite is just taking the first returned element and assuming that's the right one.

This may not be a problem in practice, since I'm assuming that the `pub_id` corresponds to a UI-created publication that wouldn't actually have multiple _task_ specs (though it would have multiple collection specs). But I'd recommend changing this function to accept a `catalog_name` parameter instead of `spec_type`, if possible.

Reply:
Cool... that was a worry I just had this morning. I checked the DB and it looks like it would be fine but still could break.

That's a good call on the `catalog_name` I'll switch to that. We have everything available because we're calling from the create/edit page.